### PR TITLE
[BREAKING] Asynchronous pin injections for Numeric Comparison and PassKey Input 

### PR DESCRIPTION
**Pull request by h2zero**
This replaces #117 as it has been rebased and includes client code updates.

Fixes https://github.com/h2zero/esp-nimble-cpp/issues/114 and https://github.com/h2zero/NimBLE-Arduino/issues/294

### CODE REVIEW

**Comment by h2zero** [comment]
@finger563 @csmith-morningstar Would you be willing to test this and provide feedback please?

**Comment by finger563** [comment]
> @finger563 @csmith-morningstar Would you be willing to test this and provide feedback please?

@h2zero can you update this branch to merge the latest `master` into it or rebase off latest `master?

**Comment by finger563** [comment]
@h2zero is there a way we can call `NimBLEDevice::injectConfirmPIN` without the conn_info? The gfps stack i'm using this in doesn't have the context of the connection (esp. not the esp-nimble-cpp NimBLEConnInfo) at the point where i need to inject confirmation/deny.

**Comment by h2zero** [comment]
The underlying stack function requires the connection id to send the pin so I can't see any other way.

**Comment by finger563** [comment]
And for reference, that code did work, so I've confirmed this works for server-side implementation of passkey injection :)

**Comment by h2zero** [comment]
Great! Thanks! 😄

File: src/NimBLEServer.h [hunk 1]
```diff
@@ -81,6 +81,8 @@ class NimBLEServer {
 #if !CONFIG_BT_NIMBLE_EXT_ADV || defined(_DOXYGEN_)
     void                   advertiseOnDisconnect(bool);
 #endif
+    void                   injectPassKey(const NimBLEAddress& address, uint32_t pin);
```
[the remark this answers is not in the document]
Comment (h2zero) [reply src/NimBLEServer.h:84]: Good catch, thanks!

**Comment by h2zero** [comment]
Thanks @csmith-morningstar, I think this is working pretty well, wouldn't want you to spend much time testing if you don't have much to spare. Thanks again for the good work you did starting this.

**Comment by finger563** [comment]
@h2zero do you expect to merge this soon, or is this going to wait longer since it's a breaking change?

**Comment by h2zero** [comment]
I will merge it later today when I can make the last edits.

**Pull request merged**